### PR TITLE
[feat] Add an adaptive subsampler

### PR DESCRIPTION
**Pull request by nabenabe0928**
Please check [the issue#426](https://github.com/automl/Auto-PyTorch/issues/426).

## Types of changes

- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)

Note that a Pull Request should only contain one of refactoring, new features or documentation changes.
Please separate these changes and send us individual PRs for each.
For more information on how to create a good pull request, please refer to [The anatomy of a perfect pull request](https://medium.com/@hugooodias/the-anatomy-of-a-perfect-pull-request-567382bb6067).

## Checklist:

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
* [ ] Have you checked to ensure there aren't other open [Pull Requests](../../../pulls) for the same update/change?
* [ ] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you written new tests for your core changes, as applicable?
* [ ] Have you successfully ran tests with your changes locally?

### CODE REVIEW

File: autoPyTorch/api/base_task.py [hunk 1]
```diff
@@ -368,6 +369,7 @@ def get_dataset(
         resampling_strategy_args: Optional[Dict[str, Any]] = None,
         dataset_name: Optional[str] = None,
         dataset_compression: Optional[DatasetCompressionSpec] = None,
+        adaptive_memory_allocation: bool = True
```
Comment (ravinkohli) [inline autoPyTorch/api/base_task.py:372]: I think its better if we allow the users to pass `data_compression = {'memory_allocation': 'adaptive' or '0.1'(example), 'methods': ['precision', 'subsample']}`. We can also do this when we finally merge it to the dev branch. Let me know if you would like to do it for this PR.

Comment (nabenabe0928) [reply autoPyTorch/api/base_task.py:372]: I will adapt the code to your suggestion quickly

Comment (ravinkohli) [reply autoPyTorch/api/base_task.py:372]: yeah and you can make `adaptive` as the default value

**Comment by ravinkohli** [comment]
@nabenabe0928 as we now have #444, do you think we can close this pull request?

**Pull request closed without merging**